.github/actions/download.sh

```diff
@@ -10,16 +10,33 @@
 # - use download-all target to get all files when package has multiple (arch specific) files.
 
 set -o pipefail
+set -e  # Exit on any command failure
+
+# Function to download a package and handle errors
+download_package() {
+    local package_dir="$1"
+    local target="$2"
+    
+    echo "===> Attempting to download: ${package_dir}"
+    if ! make -C "${package_dir}" "${target}"; then
+        echo "::error::Failed to download ${package_dir}. Exiting."
+        exit 1
+    fi
+    # Verify checksum after download
+    if ! make -C "${package_dir}" checksum; then
+        echo "::error::Checksum verification failed for ${package_dir}. Exiting."
+        exit 1
+    fi
+    echo "===> Successfully downloaded: ${package_dir}"
+}
 
 # Download regular cross/* sources
 if [ -z "${DOWNLOAD_PACKAGES}" ]; then
     echo "===> No packages to download. <==="
 else
     echo "===> Download packages: ${DOWNLOAD_PACKAGES}"
-    for download in ${DOWNLOAD_PACKAGES}
-    do
-        echo "$ make -c ${download} download-all"
-        make -C ${download} download-all
+    for download in ${DOWNLOAD_PACKAGES}; do
+        download_package "${download}" "download-all"
```

Comment:
I'm wondering why calling a download sub-function?  Maybe the the set -o pipefail or similar could be added within the download makefile recipe and simply be capturing `$?` upon exit it to validate it did succeed?  That would avoid having yet another download function that calls download functions...

Reply:
I'm not sure what you mean by this. Can you elaborate on your solution?